reascripts/common/libs/PathUtil.lua

```diff
@@ -0,0 +1,70 @@
+--[[
+
+  PathUtil.lua - Path utilities for ReaScript
+
+]]--
+
+PathUtil = {}
+
+-- Apply an extension to a file path, if it doesn't already have one.
+-- Empty or nil path returns the empty string.
+PathUtil.apply_extension = function(filepath, extension)
+  if not filepath or #filepath < 1 then
+    return ""
+  end
+
+  local root, _, ext = filepath:match("([^%.]*)(%.?([^\\/%.]*))")
+
+  if not root or #root < 1 then
+    return ""
+  end
+
+  if #ext > 0 then
+    return filepath
+  end
+
+  return filepath .. '.' .. extension
+end
+
+-- Returns the given path if a full path, otherwise returns given path
+-- relative to the REAPER project resource directory.
+PathUtil.get_real_path = function(path_arg)
+  if PathUtil._is_full_path(path_arg) then
+    return path_arg
+  end
+
+  local project_path = reaper.GetProjectPath()
+  return project_path .. PathUtil._path_separator() .. path_arg
+end
+
+-- Returns the command to reveal the given path in the OS file explorer.
+-- Works for Windows and macOS. To quote a wise philosopher, "Linux is a mess."
+PathUtil.get_reveal_command = function(path_arg)
+  if EnvUtil.is_windows() then
+    return '%SystemRoot%\\explorer.exe /select,"' .. path_arg .. '"'
+  elseif EnvUtil.is_mac() then
+    return '/bin/[ -x /usr/bin/open ] && /usr/bin/open -R "' .. path_arg .. '"'
+  else
+    return '/bin/[ -x /usr/bin/xdg-open ] && /usr/bin/xdg-open "' .. path_arg .. '"'
+  end
```

Comment:
We can avoid using the shell by using `reaper.file_exists()` to check for the file's existence. Or, we can just accept that the command might fail, and use "/usr/bin/env open" (or xdg-open), which will use the system path (should work on Mac and Linux).

On NixOS, an alternative Linux distribution, only "/bin/sh" and "/usr/bin/env" exist. So, we should probably be using "/usr/bin/env" for our curl calls as well.

Reply:
leaning towards "try `env` and okay-to-fail"